### DISCUSSION TITLE
PNP giving timeout when tries to connect connect -PnPonline.

### DISCUSSION THREAD

**Discussion by SabbirAhamm** [pnp/powershell#3931] [Q&A]
![image](https://github.com/pnp/powershell/assets/118793833/4c51766c-c009-4fc1-b95b-a0ee8e286d4b)

Connect-PnPOnline -Url
[https://xxxxx-admin.sharepoint.com -UseWebLogin
WARNING: Consider using -Interactive instead, which provides better functionality. See the documentation at
https://pnp.github.io/powershell/cmdlets/Connect-PnPOnline.html#interactive-login-for-multi-factor-authentication
Connect-PnPOnline: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond. (healthsg-admin.sharepoint.com:443)

Need to know what IP address PnP online use to connect so that it can whitelisted from network.

**Comment by FuryoftheStars** [comment]
It's attempting to connect to SharePoint, so: https://learn.microsoft.com/en-us/microsoft-365/enterprise/urls-and-ip-address-ranges?view=o365-worldwide

It's going to be a matter or trial and error to figure out which ones you actually need, but definitely the *.sharepoint.com and some of the auth urls/IPs in the Microsoft 365 Common section.  Easiest way to figure it all out, though, is to have something that'll log your connections through the firewall.  Had to do this with my previous place of employment where we wanted to block certain client systems from accessing _anything_ online except specific approved sites, to include Outlook online.  Just kept going to the log to see where the next furthest point it got before the firewall blocked it until everything worked.  I think I ended up with close to a dozen+ urls and IPs by the time I was done for just Outlook online?  Been a few years, I may not remember the count correctly.
